sklearn/metrics/_pairwise_distances_reduction/_gemm_term_computer.pyx.tp

```diff
@@ -214,4 +381,56 @@ cdef class GEMMTermComputer{{name_suffix}}:
 
         return dist_middle_terms
 
+
+cdef class SparseSparseMiddleTermComputer{{name_suffix}}(MiddleTermComputer{{name_suffix}}):
+
+    def __init__(
+        self,
+        X,
+        Y,
+        ITYPE_t effective_n_threads,
+        ITYPE_t chunks_n_threads,
+        ITYPE_t dist_middle_terms_chunks_size,
+        ITYPE_t n_features,
+        ITYPE_t chunk_size,
+    ):
+        super().__init__(
+            effective_n_threads,
+            chunks_n_threads,
+            dist_middle_terms_chunks_size,
+            n_features,
+            chunk_size,
+        )
+        self.X_data, self.X_indices, self.X_indptr = self.unpack_csr_matrix(X)
+        self.Y_data, self.Y_indices, self.Y_indptr = self.unpack_csr_matrix(Y)
+
+    cdef DTYPE_t * _compute_distances_on_chunks(
+        self,
+        ITYPE_t X_start,
+        ITYPE_t X_end,
+        ITYPE_t Y_start,
+        ITYPE_t Y_end,
+        ITYPE_t thread_num,
+    ) nogil:
+        cdef:
+            DTYPE_t *dist_middle_terms = (
+                self.dist_middle_terms_chunks[thread_num].data()
+            )
+
+        _middle_term_sparse_sparse_{{name_suffix}}(
+            self.X_data,
+            self.X_indices,
+            self.X_indptr,
+            X_start,
+            X_end,
+            self.Y_data,
+            self.Y_indices,
+            self.Y_indptr,
+            Y_start,
+            Y_end,
+            dist_middle_terms,
+        )
+
```

Comment:
If you want to inspect the content of the vector being modified (i.e. `self.dist_middle_terms_chunks[thread_num]`) using `sklearn.utils._vector_sentinel.vector_to_nd_array`, it is not possible to do it in `_middle_term_sparse_sparse_{{name_suffix}}` because:
  - `vector_to_nd_array` must take a pointer to a `std::vector` as input
  - `D` in `_middle_term_sparse_sparse_{{name_suffix}}` is the pointer to the `std::vector` internal buffer and not the buffer itself; and from there the vector is inaccessible.

Hence, I suggest you to it like this:
```suggestion
            DTYPE_t *dist_middle_terms = (
                # This is the address of the vector's buffer's first element 
                self.dist_middle_terms_chunks[thread_num].data()
            )

        with gil:
            print("DEBUG Before _middle_term_sparse_sparse_{{name_suffix}}")
            print(f"DEBUG self.dist_middle_terms_chunks[{thread_num}]=")
            vec_as_np_array = vector_to_nd_array(
                # This is the address of vector's
                &self.dist_middle_terms_chunks[thread_num],
            )
            print(vec_as_np_array)

        _middle_term_sparse_sparse_{{name_suffix}}(
            self.X_data,
            self.X_indices,
            self.X_indptr,
            X_start,
            X_end,
            self.Y_data,
            self.Y_indices,
            self.Y_indptr,
            Y_start,
            Y_end,
            dist_middle_terms,
        )

        with gil:
            print("DEBUG After _middle_term_sparse_sparse_{{name_suffix}}")
            print(f"DEBUG self.dist_middle_terms_chunks[{thread_num}]=")
            vec_as_np_array = vector_to_nd_array(
               # This is the address of vector's
               &self.dist_middle_terms_chunks[thread_num],
            )
            print(vec_as_np_array)


```